File: frontend/src/services/messagingService.ts
import api from '../config/api';
import type { ApiResponse } from '../types/api';

export interface MessageThreadSummary {
  id: number;
  subject: string;
}

export interface MessageThreadDetail {
  id: number;
  messages: Array<{ id: number; body: string }>;
}

export interface MessageRequest {
  body: string;
}

export interface MessageThreadRequest {
  listingId: number;
  tenantProfileId: number;
  landlordProfileId: number;
}

export async function getThreads(): Promise<ApiResponse<MessageThreadSummary[]>> {
  const { data } = await api.get<ApiResponse<MessageThreadSummary[]>>('/api/messages/threads');
  return data;
}

export async function getThread(threadId: string): Promise<ApiResponse<MessageThreadDetail>> {
  const { data } = await api.get<ApiResponse<MessageThreadDetail>>(`/api/messages/threads/${threadId}`);
  return data;
}

export async function createThread(payload: MessageThreadRequest): Promise<ApiResponse<MessageThreadDetail>> {
  const { data } = await api.post<ApiResponse<MessageThreadDetail>>('/api/messages/threads', payload);
  return data;
}

export async function postMessage(threadId: string, payload: MessageRequest): Promise<ApiResponse<void>> {
  const { data } = await api.post<ApiResponse<void>>(`/api/messages/threads/${threadId}/messages`, payload);
  return data;
}

File: backend/src/main/java/com/sichrplace/messaging/MessagingService.java
package com.sichrplace.messaging;

import com.sichrplace.common.dto.ApiResponse;
import com.sichrplace.messaging.dto.MessageRequest;
import com.sichrplace.messaging.dto.MessageThreadRequest;
import com.sichrplace.messaging.dto.MessageThreadResponse;
import java.util.List;

public interface MessagingService {
    ApiResponse<List<MessageThreadResponse>> listThreads();
    ApiResponse<MessageThreadResponse> getThread(Long id);
    ApiResponse<MessageThreadResponse> createThread(MessageThreadRequest request);
    ApiResponse<Void> postMessage(Long threadId, MessageRequest request);
}
